pkg/types/vsphere/validation/platform.go

```diff
@@ -38,6 +40,24 @@ func ValidatePlatform(p *vsphere.Platform, fldPath *field.Path) field.ErrorList
 	// If all VIPs are empty, skip IP validation.  All VIPs are required to be defined together.
 	if strings.Join([]string{p.APIVIP, p.IngressVIP}, "") != "" {
 		allErrs = append(allErrs, validateVIPs(p, fldPath)...)
+		if network != nil {
+			apiVIPOverlap := false
+			ingressVIPOverlap := false
+			for _, machine := range network.MachineNetwork {
```

Comment:
Performance is not going to be an issue here (MachineNetwork will be a very small array) so I would opt for readability rather than optimizing performance. I would suggest losing the boolean flags like `apiVIPOverlap` and creating a helper function that returns bool if the MachineNetwork contains the VIP